src/async_fuse/memfs/cache/backend.rs

```diff
@@ -0,0 +1,493 @@
+            })
+            .read(block.make_mut_slice())
+            .await
+        {
+            if e.kind() == std::io::ErrorKind::NotFound {
+                None
+            } else {
+                panic!(
+                    "Failed to load a block from backend where ino={ino} and block={block_id}: {e}"
+                );
+            }
+        } else {
+            Some(block)
+        }
+    }
+
+    async fn load_from_backend(&self, _: INum, _: usize) -> Option<Block> {
+        // This storage has no backend.
+        None
+    }
+
+    async fn cache_block_from_backend(&self, _: INum, _: usize, _: Block) {
+        unreachable!("This storage has no backend, and has no cache.");
+    }
+
+    async fn store(&self, ino: INum, block_id: usize, block: Block) {
+        let path = Self::get_block_path(ino, block_id);
+
+        let block_start = block.start();
+        let block_end = block.end();
+
+        if block_start == 0 && block_end == self.block_size {
+            // To store a whole block
+            let mut writer = self.operator.writer(&path).await.unwrap_or_else(|e| {
+                panic!(
+                    "Failed to get a writer to backend where ino={ino} and block={block_id}: {e}"
+                )
+            });
+            writer.write_all(block.as_slice()).await.unwrap_or_else(|e| panic!("Failed to store a block to backend where ino={ino} and block={block_id}: {e}"));
+            writer.close().await.unwrap_or_else(|e| {
+                panic!("Failed to close the writer where ion={ino} and block={block_id}: {e}")
+            });
+            return;
+        }
+
+        let mut dest = self.operator.read(&path).await.unwrap_or_else(|e| {
+            if e.kind() == ErrorKind::NotFound {
+                // Create an empty block for overwriting is ok.
+                vec![]
+            } else {
+                panic!("Failed to load a block to backend for overwrite where ino={ino} and block={block_id}: {e}");
+            }
+        });
+
+        // Ensure that the vector is long enough to be overwritten
+        if dest.len() < block_end {
+            dest.resize(block_end, 0);
+        }
+
+        // merge two blocks
+        dest.get_mut(block_start..block_end)
+            .unwrap_or_else(|| unreachable!("The vector is ensured to be long enough."))
+            .copy_from_slice(block.as_slice());
+        self.operator.write(&path, dest).await.unwrap_or_else(|e| {
+            panic!("Failed to store a block to backend where ino={ino} and block={block_id}: {e}")
+        });
+    }
+
+    async fn remove(&self, ino: INum) {
+        self.operator
+            .remove_all(&Self::get_file_path(ino))
+            .await
+            .unwrap_or_else(|e| panic!("Failed to remove a file of ino={ino}: {e}"));
+    }
+
+    async fn invalidate(&self, _: INum) {
+        // This storage has no cache, therefore, its contents cannot be
```

Comment:
The same question for these three functions as above.

Reply:
Each storage layer will also calls `invalidate()` method of its backend (see the implementation of `MemoryCache::invalidate()`), therefore, this method is always called too, so I can't raise a panic here.